tests/integration/security/mtls_cert_securenaming/mtls_cert_securenaming.go

```diff
@@ -125,9 +106,124 @@ spec:
 			}
 			checker.CheckOrFail(ctx)
 			t.Log("mTLS check passed")
+
+			t.Log("Secure naming testing #1: connection fails when DR doesn't match SA.")
+			defaultIdentityDR := fmt.Sprintf(`apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.%s.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/%s/sa/default"
+`, testNamespace.Name(), testNamespace.Name())
+			g.ApplyConfigOrFail(t, testNamespace, defaultIdentityDR)
+			time.Sleep(time.Second * 3) // Wait for the DR to take effect
```

Comment:
we should not sleep in tests if possible. Instead can we retry until successful? For something like the negative test case we can retry multiple times and assert it fails many times in a row. The retry.Converge option handles this

Reply:
Done.